packages/block-editor/src/components/list-view/block.js

```diff
@@ -103,11 +103,8 @@ function ListViewBlock( {
 	}, [ clientId, setIsHovered, highlightBlock ] );
 
 	const selectEditorBlock = useCallback(
```

Comment:
FYI this function is being called with `id` argument in here https://github.com/WordPress/gutenberg/blob/9f73be0aad5ac81e4798aef9666849b3e7ed3e0c/packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js#L83

Reply:
The problem is this also gets used here:
https://github.com/WordPress/gutenberg/blob/9543b72e701a452858b1434977275bc3fbf3a8bd/packages/block-editor/src/components/list-view/block.js#L179

And if you drill down there's no `clientId` param, it's just an `onClick` event handler:
https://github.com/WordPress/gutenberg/blob/9543b72e701a452858b1434977275bc3fbf3a8bd/packages/block-editor/src/components/list-view/block-select-button.js#L75

Reply:
I think it might be better to remove the `clientId` arg when `__experimentalSelectBlock` is called in `BlockSettingsDropdown`.

Then the only other thing that would be needed to fix this bug is to make event an optional param by adding a `?` in front of `stopPropagation`:
```jsx
	const selectEditorBlock = useCallback(
		( event ) => {
			event?.stopPropagation();
			selectBlock( clientId );
		},
		[ clientId, selectBlock ]
	);
```

Reply:
> Then the only other thing that would be needed to fix this bug is to make event an optional param by adding a `?` in front of `stopPropagation`:

That will result in an error when `selectEditorBlock` is called by `BlockSettingsDropdown` and passed a client ID as the first argument. `event` will be defined, but it will be a string that doesn't contain any function named `stopPropagation`

I think it would be clearest to quit trying to use the one callback in both `ListViewBlockContents` and `BlockSettingsDropdown`, as the former requires a prop of the form `( Event ) => void` and the latter a prop of rate form `( string ) => void`. They're incompatible.

```diff
diff --git a/packages/block-editor/src/components/list-view/block.js b/packages/block-editor/src/components/list-view/block.js
index a7746463c5..43458c3acf 100644
--- a/packages/block-editor/src/components/list-view/block.js
+++ b/packages/block-editor/src/components/list-view/block.js
@@ -102,14 +102,6 @@ function ListViewBlock( {
 		highlightBlock( clientId, false );
 	}, [ clientId, setIsHovered, highlightBlock ] );
 
-	const selectEditorBlock = useCallback(
-		( event ) => {
-			event.stopPropagation();
-			selectBlock( clientId );
-		},
-		[ clientId, selectBlock ]
-	);
-
 	const toggleExpanded = useCallback(
 		( event ) => {
 			event.stopPropagation();
@@ -176,7 +168,10 @@ function ListViewBlock( {
 					<div className="block-editor-list-view-block__contents-container">
 						<ListViewBlockContents
 							block={ block }
-							onClick={ selectEditorBlock }
+							onClick={ ( event ) => {
+								event.stopPropagation();
+								selectBlock( clientId );
+							} }
 							onToggleExpanded={ toggleExpanded }
 							isSelected={ isSelected }
 							position={ position }
@@ -235,7 +230,9 @@ function ListViewBlock( {
 								onFocus,
 							} }
 							disableOpenOnArrowDown
-							__experimentalSelectBlock={ selectEditorBlock }
+							__experimentalSelectBlock={ () =>
+								selectBlock( clientId )
+							}
 						/>
 					) }
 				</TreeGridCell>
```

Reply:
> That will result in an error when selectEditorBlock is called by BlockSettingsDropdown and passed a client ID as the first argument. event will be defined, but it will be a string that doesn't contain any function named stopPropagation

That's why I said we should remove the `clientId` argument. 😄 

> I think it would be clearest to quit trying to use the one callback in both ListViewBlockContents and BlockSettingsDropdown, as the former requires a prop of the form ( Event ) => void and the latter a prop of the form ( string ) => void. They're incompatible.

That does also seem like a good option, but we'd still have the `clientId` being passed from `BlockSettingsDropdown` to `selectBlock`. Seems like we both agree that it could be removed, or maybe your code snippet had an error in omitting the parameter?

It'd be good to use `useCallback` too, as this is performance sensitive code.

Reply:
I think this might be the right solution - https://github.com/WordPress/gutenberg/pull/38760#issuecomment-1038692197.